src/VirtoCommerce.Platform.Caching/Redis/RedisPlatformMemoryCache.cs

```diff
@@ -1,47 +1,98 @@
 using System;
+using System.Collections.Generic;
+using Microsoft.ApplicationInsights;
 using Microsoft.Extensions.Caching.Memory;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using Newtonsoft.Json;
-using Polly;
-using Polly.Retry;
 using StackExchange.Redis;
 using VirtoCommerce.Platform.Caching;
+using VirtoCommerce.Platform.Core.Caching;
 using VirtoCommerce.Platform.Core.Common;
 
 namespace VirtoCommerce.Platform.Redis
 {
     public class RedisPlatformMemoryCache : PlatformMemoryCache
     {
         private readonly ISubscriber _bus;
+        private readonly IMemoryCache _memoryCache;
+        private readonly CachingOptions _cachingOptions;
         private readonly RedisCachingOptions _redisCachingOptions;
         private readonly ILogger _log;
-        private readonly RetryPolicy _retryPolicy;
+        private readonly TelemetryClient _telemetryClient;
 
         private static readonly string _cacheId = Guid.NewGuid().ToString("N");
         
 
         public RedisPlatformMemoryCache(IMemoryCache memoryCache, IOptions<CachingOptions> options
+            , ConnectionMultiplexer connection
             , ISubscriber bus
+            , IOptions<CachingOptions> cachingOptions
             , IOptions<RedisCachingOptions> redisCachingOptions
             , ILogger<RedisPlatformMemoryCache> log
+            , TelemetryClient telemetryClient
             ) : base(memoryCache, options, log)
        {
+            _memoryCache = memoryCache;
             _log = log;
+            _telemetryClient = telemetryClient;
             _bus = bus;
 
+            _cachingOptions = cachingOptions.Value;
             _redisCachingOptions = redisCachingOptions.Value;
-            _bus.Unsubscribe(_redisCachingOptions.ChannelName);
-            _bus.Subscribe(_redisCachingOptions.ChannelName, OnMessage);
+
+            connection.ConnectionFailed += OnConnectionFailed;
+            connection.ConnectionRestored += OnConnectionRestored;
```

Comment:
subscription without unsubscription is the potential dangerous reason for memory leaking

Reply:
@t13ka Fixed. I was sure I did it, so may be @tatarincev was right and I missed to upload little part of PR. Now looks good for me.

But really we should unsubscribe here just because this class can be reused someone. It's singleton, so unsubscription isn't required here strongly.